tests/test_theming.py

```diff
@@ -62,6 +62,19 @@ def test_theme_api(app, status, warning):
     assert not os.path.exists(themedir)
 
 
+def test_no_theme_conf():
```

Comment:
Instead of creating temporary files like this, there is a fixture doing it for you (don't remember its name now but it's something like tmp_path or tmp_something, just look at some other tests or in conftest or in sphinx/testing.py).

Also, testroots should not be modified like that because they might affect other tests. 

I would suggest doing the following instead:

- use a dummy application to create a theme factory object (otherwise mypy may complain)
- create a temporary empty directory
- call `Theme(name, tmpdir, factory)`.
- catch the exception.

That way, you have no side effects and you don't need to import THEMECONF.

Btw I see that the ThemeFactory checks that THEMECONF exists when loading the external themes (lines 215 and 222 of theming.py) so I am actually wondering: when are you getting your error? I don't much about themes but does the application actually calls the Theme constructor or is it via a factory that it actually creates themes? if the Theme class is actually part of the private API, we might have assumed that the THEMECONF file actually exists when calling the constructor.

Reply:
The error occurred when I changed the `MANIFEST.in` in my theme and as a result inadvertently excluded `theme.conf` from the built artifacts (tarball, wheel). Then a documentation build using the theme failed on `sphinx-themes.org` because of the missing file. When I issued a new release after fixing my `MANIFEST.in`, the build now works. The stack trace showed me where the problem was in `Theme.__init__().`

Reply:
Ah yes, I see. Then you can directly use the Theme class in the tests (but still, change the way that it is tested)

Reply:
OK, done.

Reply:
There are some test errors in the GitHub Actions but they don't appear to be related to my changes.

Reply:
Those are flaky tests, don't worry. Btw, can you assert the exception message as well?

Reply:
Actually, I didn't assert the exception message because I wasn't sure of the i18n/l10n approach to be used - by which I mean, the exception message text is potentially dependent on the environment's locale.